sqlglot/dialects/snowflake.py

```diff
@@ -1834,8 +1844,24 @@ def alterset_sql(self, expression: exp.AlterSet) -> str:
             return f"SET{exprs}{file_format}{copy_options}{tag}"
 
         def strtotime_sql(self, expression: exp.StrToTime):
+            # Try to get type from type annotation first, fall back to target_type arg
+            target_type = None
+            if expression.type:
+                target_type = expression.type.this
+            else:
+                target_type = expression.args.get("target_type")
```

Comment:
Although the `expression.type` idea here seems interesting, it is best if we don't preemptively add extra logic, at least not without having an example / test case for it; It might be innocent but (1) the PR scope creeps up and (2) tech debt is accumulated.

If we want to keep this around, lets add tests for it and can inline it for simplicity:

```suggestion
            target_type = expression.type.this if expression.type else  expression.args.get("target_type")
```